File: Backend Web/app/Models/UMKM.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;

class UMKM extends Authenticatable
{
    use HasFactory;
    protected $table = 'umkm';

    protected $fillable = [
        'nama_umkm',
        'email',
        'password',
        'phone',
        'qris_image',
        'status',
        'user_id',
    ];
    protected static function booted()
    {
        static::creating(function ($umkm) {
            if (is_null($umkm->user_id)) {
                $umkm->user_id = 1; // Default value
            }
        });
    }

    protected $hidden = [
        'password',
        'remember_token',
    ];

    protected $casts = [
        'status' => 'boolean',
    ];

    protected $appends = ['qris_url'];

    public function products()
    {
        return $this->hasMany(Product::class, 'umkm_id');
    }

    public function getQrisUrlAttribute()
    {
        if (!$this->qris_image) return null;

        if (filter_var($this->qris_image, FILTER_VALIDATE_URL)) {
            return $this->qris_image;
        }

        return asset('storage/'.$this->qris_image);
    }
}

File: Frontend Mobile/lib/models/umkm.dart
class Umkm {
  final int id;
  final String namaUmkm;
  final String email;
  final String phone;
  final String qrisImage;
  final bool status;
  final String qrisUrl;

  Umkm({
    required this.id,
    required this.namaUmkm,
    required this.email,
    required this.phone,
    required this.qrisImage,
    required this.status,
    required this.qrisUrl,
  });

  factory Umkm.fromJson(Map<String, dynamic> json) {
    return Umkm(
      id: json['id'] as int? ?? 0,
      namaUmkm: json['nama_umkm'] as String? ?? '',
      email: json['email'] as String? ?? '',
      phone: json['phone'] as String? ?? '',
      qrisImage: json['qris_image'] as String? ?? '',
      status: json['status'] as bool? ?? false,
      qrisUrl: json['qris_url'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'nama_umkm': namaUmkm,
        'email': email,
        'phone': phone,
        'qris_image': qrisImage,
        'status': status,
        'qris_url': qrisUrl,
      };
}